importer/tests.py

```
@@ -1,3 +1,538 @@
+        self.assertEqual(result[1]["name"], "subdir")
+        self.assertEqual(len(result[1]["children"]), 1)
+        self.assertEqual(result[1]["children"][0]["name"], "nested.txt")
+        self.assertEqual(result[1]["children"][0]["children"], [])
+
+    @patch("importer.views.directory_contents")
+    def test_symlink_cycle_detection(self, mock_directory_contents):
+        """
+        Test that symlink cycles are detected using the visited set.
+
+        When build_directory_tree processes a directory whose contents include
+        a symlink back to a directory already in the visited set, the visited‑set
```

Ruff (0.15.0)

[warning] Docstring contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)